### ISSUE TITLE
Retain data found in the ID and INFO columns of the input VCF file.

### ISSUE THREAD

**Issue by sripakpa** [mskcc/vcf2maf#67]
The MAF outfile generated by the vcf2maf.pl script currently does not retain the data found in the ID column of the input VCF file.

We were wondering if it would be possible for have the vcf2maf.pl script output this data as a new column in the out MAF file.

Note: From email exchange with Dr. Cyriac Kandoth, we may additionally want to retain all the data found in the INFO column of the input VCF file as well (currently only specific data such as GT and DP are retained).

**Comment by ckandoth** [comment]
Thanks Parin. Before we start, can you give me an example of what you want to retain from a VCF's ID column? Note that VEP annotation will populate columns "dbSNP_RS" and "Existing_variation" in the MAF, which list IDs from several databases like COSMIC, dbSNP, ClinVar, etc.

**Comment by sripakpa** [comment]
Hi Cyriac,

We are hoping to use the ID column to store our internal variant_ID (integer). Here is a mock example input VCF:

##fileformat=VCFv4.2
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO
1	11312344	41489	T	G	.	.	.
1	12513166	31034	G	C	.	.	.
1	17231259	83130	C	G	.	.	.

The rough idea is that we have assigned each variant an unique variant_ID. By retaining this unique variant_ID in the output MAF file, we can then use it to look up and merge the MAF outfile with additional data in a downstream step.

**Comment by ckandoth** [comment]
I see. Have you considered the `--retain-cols` feature in maf2maf? It lets you retain data from an input tab-delimited file, in the re-annotated output MAF.

maf2maf is also good at handling the various non-standard tab-delimited formats that people invent. For example, the format below uses VCF-like alleles in a MAF-like file:
```
Chromosome  Start_Position  Reference_Allele     Tumor_Seq_Allele2  Tumor_Sample_Barcode
14          105246539       TCTTGATGTA           T                  sample_blah
7           55242464        AGGAATTAAGAGAAGCAAC  A                  sample_bleh
```
`maf2maf` will remove the VCF-style prefixed bps, increment the `Start_Position` and generate a properly formatted re-annotated MAF like this:
```
Chromosome  Start_Position  Reference_Allele     Tumor_Seq_Allele2  Tumor_Sample_Barcode
14          105246540       CTTGATGTA            -                  sample_blah
7           55242465        GGAATTAAGAGAAGCAAC   -                  sample_bleh
```

**Comment by sripakpa** [comment]
Hi Cyriac,

I see. Yes, this may work for our use-case. I will try it out tomorrow.

**Comment by ckandoth** [comment]
Commit 128932c adds 2 new MAF columns for VCF ID and QUAL. But I strongly recommend you try maf2maf - assuming it simplifies the workflow you described. The minimal required columns are explained in [the documentation](https://github.com/mskcc/vcf2maf#maf2maf). And `--retain-cols` let's you retain any columns from the input file.

If you have a variant calling workflow that produces VCFs merged from multiple variant callers, then vcf2maf is recommended. I'll continue working on this issue and implement the preservation of VCF INFO data into MAF columns.

**Comment by sripakpa** [comment]
Hi Cyriac,

Thank you for implementing this! We will test it out and let you know how it goes :).

Sincerely,
Parin S.

**Comment by ckandoth** [comment]
Please first test maf2maf for your use-case and let me know how that goes. Feel free to reach out directly over Google hangouts - https://google.com/+CyriacKandoth

**Comment by sripakpa** [comment]
Hi Cyriac,

I had the chance to try out the --retain-cols feature of the maf2maf.pl script. What I found was that this feature works for SNP variants, but will give an empty field for rows that correspond to INDEL variants. I suspect that perhaps the code is internally trying to match the variants using the REF/ALT fields. However, since the input MAF have VCF-like alleles, the match fails?

**Comment by ckandoth** [comment]
This used to be an issue in an older release of vcf2maf. Can you confirm that you are using v1.6.7, and also send me a sample input file with INDELs that I can test with? Also, why don't we have a quick phone chat or hangout? I'm interested in how exactly vcf2maf fits in your workflow. Ethan has my contact info.

**Comment by sripakpa** [comment]
Hi Cyriac,

I download vcf2maf a few days ago from:
[https://github.com/mskcc/vcf2maf/archive/master.zip](url)

Here is the sample input file with INDEL that I have tested:
[https://drive.google.com/open?id=0B0-e6kHKPsMDaTdRTVNJdTBfMms](url)

Here is the output from maf2maf that I got:
[https://drive.google.com/open?id=0B0-e6kHKPsMDLW9vR3dUNHdLYzg](url)

Yes, lets definitely chat. Would google hangout sometime this coming Monday afternoon work for you?

**Comment by ckandoth** [comment]
Yes, thanks. I've blocked out Monday 2:30p to 3:30p. I'll try to debug this before that.

**Comment by sripakpa** [comment]
Great! I just send out an invitation via Google Calendar. In the meantime, I will try to checkout the latest commit of the codebase and see if the issue is already resolved there. :)

**Comment by sripakpa** [comment]
Hi Cyriac:

Btw, I just confirmed that I am currently using the following commit of vcf2maf:

    commit 3f757be48e33a3e6c6750d599cb18e4da90d1d5a
    Merge: 876190d 11e9ed1
    Author: Cyriac Kandoth <ckandoth@gmail.com>
    Date:   Thu Aug 11 12:57:16 2016 -0400

        Merge pull request #64 from sbg/dockerizeit-1.6.7

        Adding a dockerfile using VEP 85, after 1.6.7 release.

This commit is newer than v1.6.7:

    git show v1.6.7
    commit 194d2349818689ca06ac6ab44b45273d6353a054
    Author: Cyriac Kandoth <ckandoth@gmail.com>
    Date:   Thu Jun 9 16:15:19 2016 -0400

        Fix typo that blocked reporting Splice_Region

**Comment by ckandoth** [comment]
Thanks @sripakpa . I've created a separate issue for that problem. Definitely a niche case I didn't test, when I added support for VCF-style indels in what are supposed to be MAF files. :)

**Comment by sripakpa** [comment]
Hi @ckandoth,

Thank you. Should I go ahead and close this ticket?

**Comment by ckandoth** [comment]
No. The `INFO` data is still pending. It's not a priority, so I'll put this on the backburner.

**Comment by ckandoth** [comment]
Turns out it was easy enough to add a new `--retain-info` feature for `vcf2maf`. Now implemented in cccebe6
